src/mobu/services/flock.py

```diff
@@ -209,11 +217,22 @@ def _create_monkey(self, user: AuthenticatedUser) -> Monkey:
     async def _create_users(self) -> list[AuthenticatedUser]:
         """Create the authenticated users the monkeys will run as."""
         users = self._config.users
+        if users and self._replica_count > 1:
+            raise RuntimeError(
+                "Only user_spec users are allowed if replica_count > 1"
+            )
```

Comment:
This should probably raise a custom exception that inherits from Safir's `ClientRequestError` so that mobu returns a 400 error if someone tries to create a flock rather than a 500 error. Or, alternately, only pass in the replica count and instance ID for autostart configurations and don't use them at all when the user manually creates a flock via the API, since they don't really make sense in that case anyway unless we're going to add Redis or something for runtime coordination of spreading monkeys across flocks (which feels like total overkill).

Reply:
I like raising a `ClientRequestError`. 

I considered disabling this handler, and other functionality that may be wonky, like the GitHub functionality. I decided it's not worth it for now and added a bunch of warnings to the docs and Phalanx config instead, but I could definitely be convinced otherwise.

Reply:
That's fine with me; we can always clean it up later.

Reply:
Sharding the monkeys across replicas gets rid of any of these restrictions on specifying users, so this doesn't apply at all any more.